### PR TITLE
Button clean-up and documentation

### PR DESCRIPTION
**Pull request by Fercas123**
Button changes in this PR:
- CSS clean. remove of old patches
- CSS add active state when space/enter are pressed
- Remove DivButton and it's implementation as it is out of scope. 
- Improve documentation

### CODE REVIEW

File: packages/core/src/pill/internal/DivButton.tsx [hunk 1]
```diff
@@ -41,14 +33,13 @@ export const DivButton = forwardRef<HTMLDivElement, DivButtonProps>(
...
       role: roleProp,
-      variant = "primary",
       ...restProps
     },
     ref?
   ): ReactElement<DivButtonProps> {
     const enter = "Enter";
     const space = " ";
-
+    //
```
Comment (joshwooding) [inline packages/core/src/pill/internal/DivButton.tsx:42]: ?

File: packages/core/stories/layout/stack-layout.stories.tsx [hunk 2]
```diff
@@ -6,7 +6,6 @@ import {
   StackLayout,
 } from "@jpmorganchase/uitk-core";
 import { SearchIcon } from "@jpmorganchase/uitk-icons";
-import { Dropdown } from "@jpmorganchase/uitk-lab";
```
Comment (joshwooding) [inline packages/core/stories/layout/stack-layout.stories.tsx:9]: Nit: Change doesn't seem relevant to Button

File: packages/lab/src/__tests__/__e2e__/contact-details/ContactDetails.accessibility.cy.tsx [hunk 3]
```diff
@@ -146,7 +146,7 @@ variants.forEach((variant) => {
       if (variant !== "mini") {
         for (const [label, text] of testActions) {
           cy.realPress("Tab");
-          cy.findByText(label).parent().should("have.focus");
+          cy.findByText(label).should("have.focus");
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/contact-details/ContactDetails.accessibility.cy.tsx:149]: 👍🏽

**Review by sion-docs — APPROVED** [review]

File: packages/core/src/button/useButton.ts [hunk 4]
```diff
@@ -82,7 +82,7 @@ export const useButton = <T extends Element>({
   };
 
   const buttonProps = {
-    "aria-disabled": disabled,
+    "aria-disabled": focusableWhenDisabled,
```
Comment (joshwooding) [inline packages/core/src/button/useButton.ts:85]: Not sure this is right, I would expect aria-disabled to be on when disabled and focusWhenDisabled is true

File: packages/core/src/button/useButton.ts [hunk 5]
```diff
@@ -82,7 +82,7 @@ export const useButton = <T extends Element>({
   };
 
   const buttonProps = {
-    "aria-disabled": disabled,
+    "aria-disabled": disabled && focusableWhenDisabled,
```
Comment (joshwooding) [inline packages/core/src/button/useButton.ts:85]:
```suggestion
    "aria-disabled": disabled && focusableWhenDisabled ? true : undefined,
```
Sorry, I think this should be this. This should prevent aria-disabled being applied when the button isn't disabled

**Review by joshwooding — CHANGES_REQUESTED** [review]
See comment above, sorry!

**Review by joshwooding — DISMISSED** [review]

File: .changeset/warm-mangos-brake.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+"@jpmorganchase/uitk-core": minor
+"@jpmorganchase/uitk-lab": minor
+---
+
+### Button refactor
+
+- Removed internal `<span/>` from `Button`, children are now direct children of the component.
+- CSS Refactor to remove old browser support.
+- fix `aria-disabled` to only show when not redundant.
```
Comment (joshwooding) [inline .changeset/warm-mangos-brake.md:10]:
```suggestion
- Fix `aria-disabled` to only show when not redundant.
```

File: .changeset/warm-mangos-brake.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+- Added active visual style to `Button` when either `spacebar` or `enter`.
+- Removed of `DivButton`. for lab components that used `DivButton`, it's been replaced by a `Button` (`Tab` and `Dropdown`) or a custom internal `div` (`Pill`).
```
Comment (joshwooding) [inline .changeset/warm-mangos-brake.md:12]:
```suggestion
- Removed `DivButton`. For lab components that used `DivButton`, it's been replaced by a `Button` (`Tab` and `Dropdown`) or a custom internal `div` (`Pill`).
```

**Pull request merged**